core/provider/service.go

```diff
@@ -276,6 +277,31 @@ func (s *Service) RevokeAccess(a *domain.Appeal) error {
 	// handle the resolution for the appeal status
 }
 
+func (s *Service) Delete(id string) error {
+	p, err := s.providerRepository.GetByID(id)
+	if err != nil {
+		return fmt.Errorf("getting provider details: %w", err)
+	}
+
+	resources, err := s.resourceService.Find(map[string]interface{}{
+		"provider_type": p.Type,
+		"provider_urn":  p.URN,
+	})
+	if err != nil {
+		return fmt.Errorf("retrieving related resources: %w", err)
+	}
+	var resourceIds []string
+	for _, r := range resources {
+		resourceIds = append(resourceIds, r.ID)
+	}
+	// TODO: execute in transaction
```

Comment:
does it mean we do this in another PR?
or this one is missing in current PR?

Reply:
won't do in this PR, since there's also a big changes here https://github.com/odpf/guardian/pull/155, will do at least after both got merged